Accept a prebuilt resource compiler, which must run on the build platform.

--- a/profiler/CMakeLists.txt
+++ b/profiler/CMakeLists.txt
@@ -40,13 +40,20 @@
 include(${CMAKE_CURRENT_LIST_DIR}/../cmake/server.cmake)
 include(${CMAKE_CURRENT_LIST_DIR}/../cmake/GitRef.cmake)
 
-include(ExternalProject)
-ExternalProject_Add(embed
-    SOURCE_DIR ${CMAKE_CURRENT_LIST_DIR}/helpers
-    CMAKE_ARGS
-        -DCMAKE_BUILD_TYPE=Release
-        -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}
-)
+if(TRACY_EMBED_TOOL)
+    # Resource generation runs on the build platform, including in cross builds.
+    add_executable(embed IMPORTED)
+    set_target_properties(embed PROPERTIES IMPORTED_LOCATION "${TRACY_EMBED_TOOL}")
+else()
+    include(ExternalProject)
+    ExternalProject_Add(embed
+        SOURCE_DIR ${CMAKE_CURRENT_LIST_DIR}/helpers
+        CMAKE_ARGS
+            -DCMAKE_BUILD_TYPE=Release
+            -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}
+    )
+    set(TRACY_EMBED_TOOL ${CMAKE_CURRENT_BINARY_DIR}/embed)
+endif()
 
 function(Embed LIST NAME FILE)
     cmake_parse_arguments(EMBED "TEXT" "" "" ${ARGN})
@@ -58,7 +65,7 @@
     add_custom_command(
         OUTPUT data/${NAME}.cpp data/${NAME}.hpp
         COMMAND ${CMAKE_COMMAND} -E make_directory data
-        COMMAND ${CMAKE_CURRENT_BINARY_DIR}/embed ${EMBED_FLAGS} ${NAME} ${CMAKE_CURRENT_LIST_DIR}/${FILE} data/${NAME}
+        COMMAND ${TRACY_EMBED_TOOL} ${EMBED_FLAGS} ${NAME} ${CMAKE_CURRENT_LIST_DIR}/${FILE} data/${NAME}
         DEPENDS embed ${CMAKE_CURRENT_LIST_DIR}/${FILE}
     )
     list(APPEND ${LIST} data/${NAME}.cpp)
